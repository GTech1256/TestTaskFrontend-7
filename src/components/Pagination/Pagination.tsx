import React from 'react'


export type Props = {
  hasNextPage?: boolean
  hasPreviousPage?: boolean
  onPrev: () => void
  onNext: () => void
}

const Pagination = ({
  hasNextPage = false,
  hasPreviousPage= false,
  onPrev,
  onNext,
}: Props) => {
  return (
    <nav aria-label="Page navigation example">
      <ul className="pagination justify-content-center">
        <li className={`page-item ${!hasPreviousPage ? 'disabled' : ''}`}>
          <button
            className="page-link"
            onClick={onPrev}
            disabled={!hasPreviousPage}
          >
            <span aria-hidden="true">&laquo;</span>
            <span className="sr-only">Предыдущая</span>
          </button>
        </li>
        <li className={`page-item ${!hasNextPage ? 'disabled' : ''}`}>
          <button
            className="page-link"
            onClick={onNext}
            disabled={!hasNextPage}
          >
          <span aria-hidden="true">&raquo;</span>
          <span className="sr-only">Следующая</span>
          </button>
        </li>
      </ul>
    </nav>
  )
}

export default Pagination