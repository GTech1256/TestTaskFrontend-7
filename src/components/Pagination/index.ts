export type { Props as PaginationProps } from './Pagination'
export { default } from './Pagination'