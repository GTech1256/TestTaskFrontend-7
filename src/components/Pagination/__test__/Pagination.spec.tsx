import React from 'react';
import renderer from 'react-test-renderer';

import Pagination, { PaginationProps } from '../index'


const props: PaginationProps = {
  hasNextPage: true,
  onPrev: () => {},
  onNext: () => {},
}

describe('<Pagination />', () => {
  it('should render correct', () => {
    const tree = renderer
      .create(<Pagination {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  })
})