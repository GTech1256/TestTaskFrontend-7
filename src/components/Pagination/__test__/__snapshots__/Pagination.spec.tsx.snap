// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render correct 1`] = `
<nav
  aria-label="Page navigation example"
>
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item disabled"
    >
      <button
        className="page-link"
        disabled={true}
        onClick={[Function]}
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          className="sr-only"
        >
          Предыдущая
        </span>
      </button>
    </li>
    <li
      className="page-item "
    >
      <button
        className="page-link"
        disabled={false}
        onClick={[Function]}
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          className="sr-only"
        >
          Следующая
        </span>
      </button>
    </li>
  </ul>
</nav>
`;
